pithos/pithos.py

```diff
@@ -1039,6 +1068,7 @@ def on_prefs_response(self, widget, response):
 
         email = self.settings.get_string('email')
         if response == Gtk.ResponseType.APPLY:
+            self.worker_run(self.on_explicit_content_filter_checkbox)
```

Comment:
>  The art_worker modifies the UI from a thread?

Nope, it calls ```art_callback``` on the main thread.

> I know that's why I was using GLib.idle_add and not the worker but you told me to use the worker?

Right you can't block the UI but you can't modify the UI from a thread either, thankfully worker has this callback system that does work in a thread but calls back to the UI thread when done.